src/VisualStudio/Core/Test/Venus/DocumentServiceTests.vb

```diff
@@ -0,0 +1,245 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Threading
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.Editor.UnitTests
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Classification
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Classification.FormattedClassifications
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces
+Imports Microsoft.CodeAnalysis.Host
+Imports Microsoft.CodeAnalysis.Test.Utilities
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.Text.Shared.Extensions
+Imports Microsoft.VisualStudio.LanguageServices.Implementation.Venus
+Imports Roslyn.Test.Utilities
+
+Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.Venus
+
+    <UseExportProvider>
+    Public Class DocumentServiceTests
+        <WpfFact, Trait(Traits.Feature, Traits.Features.Venus)>
+        Public Async Function TestSpanMappingService() As Task
+            Dim service = ContainedDocument.DocumentServiceProvider.Instace
+            Dim spanMapper = service.GetService(Of ISpanMappingService)
+
+            Using workspace = TestWorkspace.Create(
+                <Workspace>
+                    <Project Language="C#" CommonReferences="true">
+                        <Document>class outter { {|Document:class C { $$ }|} }</Document>
+                    </Project>
+                </Workspace>, exportProvider:=TestExportProvider.ExportProviderWithCSharpAndVisualBasic)
+
+                Dim subjectDocument = workspace.Documents.Single()
+                Dim projectedDocument As TestHostDocument = CreateAndAddProjectionBufferDocument(workspace, subjectDocument)
+
+                Dim position = projectedDocument.CursorPosition.Value
+                Dim spans = Await spanMapper.MapSpansAsync(workspace.CurrentSolution.GetDocument(projectedDocument.Id), {New TextSpan(position, length:=0)}, CancellationToken.None)
+
+                Assert.Equal(1, spans.Count)
+                Assert.Equal(subjectDocument.CursorPosition.Value, spans(0).Span.Start)
+                Assert.Equal(projectedDocument.FilePath, spans(0).FilePath)
+            End Using
+        End Function
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.Venus)>
+        Public Async Function TestSpanMappingService_InvalidPosition() As Task
+            Dim service = ContainedDocument.DocumentServiceProvider.Instace
```

Comment:
Seems odd we're not getting this service from the workspace. This is kinda one of those things that @sharwell was trying to avoid.